### PR TITLE
Add a commit style guide

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
I could use more suggestions / amendments here.

### CODE REVIEW

File: doc/Style.md [hunk 1]
```diff
@@ -10,6 +10,43 @@ We mostly follow [the D style](https://dlang.org/dstyle.html).
 
 The following is a list of exceptions:
 
+### Commits
```
Comment (Geod24) [inline doc/Style.md]: That shouldn't be under `Formatting` though

**Comment by AndrejMitrovic** [comment]
Updated suggested changes.

File: doc/Style.md [hunk 2]
```diff
@@ -4,6 +4,45 @@ This document describes the style used within Agora.
 It covers the style itself, which should be clear and unambiguous,
 as well as more design aspects on how we approach / solve problems.
 
+## Commits
+
+- A commit must be a self-contained unit of logical change.
+
+- A commit should not contain unrelated changes mixed into the same commit.
```
Comment (Geod24) [inline doc/Style.md:11]: Kinda redundant with point 1, no ?

File: doc/Style.md [hunk 3]
```diff
@@ -4,6 +4,45 @@ This document describes the style used within Agora.
...
+
+- A commit should be relatively simple to review. If the commit changes
+too many files, or adds too much functionality, consider splitting it
+up into several distinct commits.
+
+- Each commit **must be** compileable and **must** succeed a run of the test-suite.
```
Comment (Geod24) [inline doc/Style.md]:
```suggestion
- Each commit **must** compile and **must** pass all tests. Hence it is not good form to add tests and then make them pass.
```

The "Hence [...]" part could be taken off, if you wish

File: doc/Style.md [hunk 4]
```diff
@@ -4,6 +4,45 @@ This document describes the style used within Agora.
...
+
+- The commit title and description should follow some general rules:
+
+An example good commit: https://github.com/bpfkorea/agora/commit/71e7d1cc1da314d92eb76c1d02e04f43bc83784f
```
Comment (Geod24) [inline doc/Style.md:21]: I originally wanted to just link to the article and call it a day, as it does a very good job at explaining what's needed, but I like the focus you put there 👍

**Review by Geod24 — COMMENTED** [review]
Just a few nitpicks

File: doc/Style.md (hunk 2 above)
Comment (AndrejMitrovic) [reply doc/Style.md:11]: I'm just trying to make it more accessible to non-native speakers by driving the point.

**Comment by AndrejMitrovic** [comment]
Thanks, updated.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**